packages/installer/upgrade.sh

```diff
@@ -54,10 +54,23 @@ find_recovery() {
 
 # cos-upgrade-image: system/cos
 find_upgrade_channel() {
+
+    if [ -e "/etc/environment" ]; then
+        source /etc/environment
+    fi
+
+    if [ -e "/etc/os-release" ]; then
+        source /etc/os-release
+    fi
+
     if [ -e "/etc/cos-upgrade-image" ]; then
         source /etc/cos-upgrade-image
     fi
 
+    if [ -e "/etc/cos/config" ]; then
+        source /etc/cos/config
```

Comment:
Isn't it still overwriting what was passed in the command line? `find_upgrade_channel` happens right after evaluating the command line parameters. If the command line includes `--recovery` and `/etc/cos/config` includes `UPGRADE_RECOVERY=false` the recovery flag turns to be useless. I am unsure of what is expected to be in `/etc/cos/config` though

Reply:
The image that is being passed from the CLI is being set right afterwards here: https://github.com/rancher-sandbox/cOS-toolkit/blob/f272f6e25e9121ca5a808ee45dd9f8982738bce8/packages/installer/upgrade.sh#L65

UPGRADE_RECOVERY, even if could be set here it shouldn't be part of the config, it's just to drive the CLI selection

Reply:
OH I see, this is for the image parameter and the config associated with the upgrade image. All clear now, makes complete sense.